docs/knowledgebase/getting-started/glossary.md

```diff
@@ -220,80 +233,89 @@ some arguments. Such published functions take one additional parameter, known as
+is a performance optimization that up-to-date nodes can take advantage of.
 
-### Native Executor
+### Wasm Executor
 
-An Executor that uses the current inbuilt and natively compiled runtime to execute the needed call.
-If the code is compatible with the on-chain Wasm code, then it will be a lot faster and correct.
-Incorrect versioning will result in a consensus error.
+This executor uses a [Wasm](#webassembly-wasm) binary and a Wasm interpreter to execute calls. Since
+Substrate-based chains include Wasm binaries as part of their [state](#state), this is guaranteed to
+be up-to-date regardless of the version of the blockchain [node](#node).
 
 ## Extrinsic
 
-A piece of data bundled into a block that expresses something from the "external" (i.e. off-chain)
-world. There are, broadly speaking, two types of extrinsic: transactions (which tend to be signed)
-and inherents (which don't).
+A piece of data that is bundled into a [block](#block) in order to express something from the
+"external" (i.e. off-chain) world. There are, broadly speaking, two types of extrinsics:
+transactions, which may be
+[signed](../../knowledgebase/learn-substrate/extrinsics#signed-transactions) or
+[unsigned](../../knowledgebase/learn-substrate/extrinsics#unsigned-transactions), and
+[inherents](../../knowledgebase/learn-substrate/extrinsics#inherents), which are inserted by
+[block authors](#author).
 
 ## Existential Deposit
 
-Within the SRML Balances module, this is the minimum balance an account may have. Accounts cannot be
-created with a balance less than some minimal amount, and if an account's balance ever drops below
-amount, then it is removed entirely from the system.
-
-> As long as both the sender and receiver keep a balance of at least the existential deposit,
-> transfers, including of very small amounts, will work intuitively. However, an attempt to transfer
-> a small balance from one account to another can have slightly less-than-intuitive consequences
-> when either the sender or receiver balances are very low.
->
-> If the amount transferred is less than the existential deposit and the destination account did not
-> previously exist, then the transfer will "succeed" without actually creating and crediting the
-> destination account; this appears to essentially just burn the transfer balance from the sender.
-> If the transfer takes the sender to below the existential balance, then its account will be
-> deleted. In this way, a transfer can even "successfully" result in the sender account being
-> completely deleted, with the receiver account never being created.
->
-> It is up to middleware to ensure that end-users are made aware of and/or protected from these edge
-> cases.
+Within the [Balances pallet](../../knowledgebase/runtime/frame#balances), this is the minimum
+balance an account may have. Accounts cannot be created with a balance less than this amount, and if
+an account's balance ever drops below this amount, then it is removed entirely.
```

Comment:
May want to make a note about reference counting too. Accounts are stored in the System module, and the ED is just one factor in removing accounts.